server/queues/processors/WebhookProcessor.ts

```diff
@@ -0,0 +1,383 @@
+
+    await this.sendModelWebhook({
+      event,
+      subscription,
+      modelId: event.modelId,
+      modelPayload:
+        hydratedFileOperation && presentFileOperation(hydratedFileOperation),
+    });
+  }
+
+  async handleDocumentEvent(
+    subscription: WebhookSubscription,
+    event: DocumentEvent | RevisionEvent
+  ): Promise<void> {
+    const hydratedDocument = await Document.findByPk(event.documentId);
+
+    await this.sendModelWebhook({
+      event,
+      subscription,
+      modelId: event.documentId,
+      modelPayload: hydratedDocument && presentDocument(hydratedDocument),
+    });
+  }
+
+  async handleUserEvent(
+    subscription: WebhookSubscription,
+    event: UserEvent
+  ): Promise<void> {
+    if (event.name === "users.invite") {
+      const invite = event.data;
+
+      await this.sendWebhook({
+        event,
+        subscription,
+        payload: { model: invite },
+      });
+    } else {
+      const hydratedUser = await User.findByPk(event.userId);
+
+      await this.sendModelWebhook({
+        event,
+        subscription,
+        modelId: event.userId,
+        modelPayload: hydratedUser && presentUser(hydratedUser),
+      });
+    }
+  }
+
+  async sendModelWebhook({
+    event,
+    subscription,
+    modelPayload,
+    modelId,
+  }: {
+    event: Event;
+    subscription: WebhookSubscription;
+    modelPayload: unknown;
+    modelId: string;
+  }) {
+    const payload = {
+      id: modelId,
+      model: modelPayload,
+    };
+    await this.sendWebhook({ event, subscription, payload });
+  }
+
+  async sendWebhook({
+    event,
+    subscription,
+    payload,
+  }: {
+    event: Event;
+    subscription: WebhookSubscription;
+    payload: { model: unknown; id?: string };
+  }) {
+    const delivery = await WebhookDelivery.create({
+      webhookSubscriptionId: subscription.id,
+      status: "pending",
+    });
+
+    const jsonBody = presentWebhook({
+      event,
+      delivery,
+      subscription,
+      payload,
+    });
+    const body = JSON.stringify(jsonBody);
+
+    const webhookUrl = subscription.url;
+    const headers = {
+      "Content-Type": "application/json",
+    };
+
+    const response = await fetch(webhookUrl, {
+      method: "POST",
+      headers,
+      body,
+    });
+
+    const newStatus = response.ok ? "success" : "failed";
+
+    await delivery.update({
+      status: newStatus,
+      statusCode: response.status,
+      requestBody: jsonBody,
+      requestHeaders: headers,
+      responseBody: response.body?.toString(),
+      responseHeaders: response.headers,
+    });
+
+    if (!response.ok) {
+      const recentDeliveries = await WebhookDelivery.findAll({
+        where: { webhookSubscriptionId: subscription.id },
+        order: [["createdAt", "DESC"]],
+        limit: 25,
+      });
+
+      const allFailed = recentDeliveries.every(
+        (delivery) => delivery.status === "failed"
+      );
+
+      if (allFailed) {
```

Comment:
This is actually immediately disabling the webhook on the first failure, need to check that `recentDeliveries` actually returns 25 results.

```suggestion
      if (recentDeliveries.length === 25 && allFailed) {
```

Reply:
🤦 

Nice catch! Fixed and tested